File: Kaizen.Core/Models/Schedule.cs
using Microsoft.Identity.Client;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Kaizen.Core.Models
{
    public class Schedule
    {
        public int Id { get; set; }

        public int SubjectId { get; set; }

        public Subject Subject { get; set; } = null!;

        public int GradeId { get; set; }

        public Grade Grade { get; set; } = null!;

        public int ClassRoomId { get; set; }

        public ClassRoom ClassRoom { get; set; } = null!;

        public int TimeSlotId { get; set; }

        public TimeSlot TimeSlot { get; set; } = null!;
    }
}

File: Kaizen.Ng/src/models/schedule.ts
import { ClassRoom } from "./class-room";
import { Grade } from "./grade";
import { Subject } from "./subject";
import { TimeSlot } from "./time-slot";

export class Schedule {

    id : number = 0;

    classRoomId : number = 0;
    gradeId : number = 0;
    subjectId : number = 0;
    timeSlotId : number = 0;

    classRoom : ClassRoom | null = null;
    grade : Grade | null = null;
    subject : Subject | null = null;
    timeSlot : TimeSlot | null = null;
}